dsp_permissions_scripts/oap/oap_get.py

```diff
@@ -112,17 +112,21 @@ def _get_value_oaps(resource: dict[str, Any], restrict_to_props: list[str] | Non
             continue
         if restrict_to_props is not None and k not in restrict_to_props:
             continue
-        match v:
-            case {
-                "@id": id_,
-                "@type": type_,
-                "knora-api:hasPermissions": perm_str,
-            } if "/values/" in id_:
-                scope = create_scope_from_string(perm_str)
-                oap = ValueOap(scope=scope, property=k, value_type=type_, value_iri=id_, resource_iri=resource["@id"])
-                res.append(oap)
-            case _:
-                continue
+        values = v if isinstance(v, list) else [v]
+        for val in values:
+            match val:
+                case {
```

Comment:
I dislike that you match something and construct it at the same time. If that is not what is happening, then I really don't understand this...

Reply:
Could you further elaborate the problem? I admit that these lines of code are rather complex, but I don't see a fundamental problem.

Reply:
I am not saying it is a problem. I am just saying that it is really hard to understand.